src/core/thread/mle_router.cpp

```diff
@@ -2040,11 +2040,33 @@ otError MleRouter::UpdateChildAddresses(const Message &aMessage, uint16_t aOffse
     uint16_t                 offset          = 0;
     uint16_t                 end             = 0;
 
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+    Ip6::Address oldMlrRegisteredAddresses[OPENTHREAD_CONFIG_MLE_IP_ADDRS_PER_CHILD - 1];
+    uint16_t     oldMlrRegisteredAddressNum = 0;
+#endif
+
     VerifyOrExit(aMessage.Read(aOffset, sizeof(tlv), &tlv) == sizeof(tlv), error = OT_ERROR_PARSE);
     VerifyOrExit(tlv.GetLength() <= (aMessage.GetLength() - aOffset - sizeof(tlv)), error = OT_ERROR_PARSE);
 
     offset = aOffset + sizeof(tlv);
     end    = offset + tlv.GetLength();
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+    // Retrieve registered multicast addresses of the Child
+    if (aChild.HasAnyMlrRegisteredAddress())
+    {
+        auto childAddresses = aChild.IterateIp6Addresses(Ip6::Address::kTypeMulticastLargerThanRealmLocal);
+
+        for (auto iter = childAddresses.begin(); iter != childAddresses.end(); iter++)
```

Comment:
Suggest using the iterator directly to avoid using hidden/private type (through use of `auto`).

```
for (Child::AddressIterator iter(aChild, Ip6::Address::kTypeMulticastLargerThanRealmLocal); !iter.IsDone(); iter++)
```

I think we want to allow `IterateIp6Addresses()`  for use in range-based `for` loop only.

Reply:
I think there is no such code now.